ci_pod/flask/app/app.py

```diff
@@ -0,0 +1,95 @@
+import os
+import shutil
+import boto3
+from flask import Flask, send_file, abort
+
+app = Flask(__name__)
+
+
+@app.route('/health')
+def health():
+    """
+    This method run health check
+    @return:
+    """
+    return 'OK'
+
+
+@app.route('/', defaults={'path': ''})
+@app.route('/<path:path>')
+def flask_download_file(path):
+    """
+    This method get path and download file from s3
+    @param path:
+    @return:
+    """
+    temp_dir_name = '/tmp/pod_ci_data'
+    
+    # prepare storage folder
+    os.makedirs(temp_dir_name, exist_ok=True)
+    # show the path after /
+    temp = (str(path)).split("/")
+    file_name = temp[(len(temp) - 1)]
+    list_folder = temp[:-1]
+    folder = ("/").join(list_folder)
+    store_path = os.path.join(temp_dir_name, folder)
+    os.makedirs(store_path, exist_ok=True)
+    s3path = str(path)
+    local_path = os.path.join(store_path, file_name)
+
+    # read data from file due to flask limitation to read from environment variable
+    bucket = file_read_string("/tmp/BUCKET")
+    aws_default_region = file_read_string("/tmp/AWS_DEFAULT_REGION")
+    endpoint_url = file_read_string("/tmp/ENDPOINT_URL")
+    aws_access_key_id = file_read_string("/tmp/AWS_ACCESS_KEY_ID")
+    aws_secret_access_key = file_read_string("/tmp/AWS_SECRET_ACCESS_KEY")
+    try:
+        s3_client = boto3.client(service_name='s3',
+                                 region_name=aws_default_region,
+                                 endpoint_url=endpoint_url,
+                                 aws_access_key_id=aws_access_key_id,
+                                 aws_secret_access_key=aws_secret_access_key)
+        s3_client.download_file(Bucket=bucket, Key=s3path, Filename=local_path)
+    except Exception as e:
+        print("Exception:")
+        print(str(e))
+        abort(404)
+
+    result = send_file(local_path, as_attachment=True)
+    # remove local folder
+    shutil.rmtree(store_path)
+    return result
+
+
+# mode w for overwrite
+def file_add_string(path, string, mode="a"):
+    """
+    This method write to file
+    @param path:
+    @param string:
+    @param mode:
+    @return:
+    """
+    with open(path, mode) as myfile:
+        myfile.write(string)
+
+
+def file_read_string(path):
+    """
+    This method read file
+    @param path:
+    @return:
+    """
+    with open(path, "r") as myfile:
+        return myfile.read()
+
+
+if __name__ == "__main__":
+
+    # Important: Store environment variables in file due to flask did not recognized environment variables
+    file_add_string("/tmp/BUCKET", os.environ['BUCKET'], "w")
+    file_add_string("/tmp/AWS_DEFAULT_REGION", os.environ['AWS_DEFAULT_REGION'], "w")
+    file_add_string("/tmp/ENDPOINT_URL", os.environ['ENDPOINT_URL'], "w")
+    file_add_string("/tmp/AWS_ACCESS_KEY_ID", os.environ['AWS_ACCESS_KEY_ID'], "w")
+    file_add_string("/tmp/AWS_SECRET_ACCESS_KEY", os.environ['AWS_SECRET_ACCESS_KEY'], "w")
```

Comment:
Use a temporary directory under /tmp (via mktemp() or similar) rather than top level inside /tmp (which is world readable).